core/chains/evm/txmgr/orm_test.go

```diff
@@ -361,3 +365,168 @@ func TestORM_UpdateBroadcastAts(t *testing.T) {
 		assert.Equal(t, etx.BroadcastAt.Year(), time2.Year())
 	})
 }
+
+func TestORM_SetBroadcastBeforeBlockNum(t *testing.T) {
+	t.Parallel()
+
+	db := pgtest.NewSqlxDB(t)
+	cfg := newTestChainScopedConfig(t)
+	borm := cltest.NewTxmORM(t, db, cfg)
+	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
+	ethClient := evmtest.NewEthClientMockWithDefaultChain(t)
+	_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
+	etx := cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, borm, 0, fromAddress)
+	chainID := *ethClient.ChainID()
+
+	headNum := int64(9000)
+	var err error
+
+	t.Run("saves block num to unconfirmed eth_tx_attempts without one", func(t *testing.T) {
+		// Do the thing
+		require.NoError(t, borm.SetBroadcastBeforeBlockNum(headNum, chainID))
+
+		etx, err = borm.FindEthTxWithAttempts(etx.ID)
+		require.NoError(t, err)
+		require.Len(t, etx.EthTxAttempts, 1)
+		attempt := etx.EthTxAttempts[0]
+
+		assert.Equal(t, int64(9000), *attempt.BroadcastBeforeBlockNum)
+	})
+
+	t.Run("does not change eth_tx_attempts that already have BroadcastBeforeBlockNum set", func(t *testing.T) {
+		n := int64(42)
+		attempt := newBroadcastLegacyEthTxAttempt(t, etx.ID, 2)
+		attempt.BroadcastBeforeBlockNum = &n
+		require.NoError(t, borm.InsertEthTxAttempt(&attempt))
+
+		// Do the thing
+		require.NoError(t, borm.SetBroadcastBeforeBlockNum(headNum, chainID))
+
+		etx, err = borm.FindEthTxWithAttempts(etx.ID)
+		require.NoError(t, err)
+		require.Len(t, etx.EthTxAttempts, 2)
+		attempt = etx.EthTxAttempts[0]
+
+		assert.Equal(t, int64(42), *attempt.BroadcastBeforeBlockNum)
+	})
+
+	t.Run("only updates eth_tx_attempts for the current chain", func(t *testing.T) {
+		require.NoError(t, ethKeyStore.Enable(fromAddress, testutils.SimulatedChainID))
+		etxThisChain := cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, borm, 1, fromAddress, cfg.DefaultChainID())
+		etxOtherChain := cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, borm, 0, fromAddress, testutils.SimulatedChainID)
+
+		require.NoError(t, borm.SetBroadcastBeforeBlockNum(headNum, chainID))
+
+		etxThisChain, err = borm.FindEthTxWithAttempts(etxThisChain.ID)
+		require.NoError(t, err)
+		require.Len(t, etxThisChain.EthTxAttempts, 1)
+		attempt := etxThisChain.EthTxAttempts[0]
+
+		assert.Equal(t, int64(9000), *attempt.BroadcastBeforeBlockNum)
+
+		etxOtherChain, err = borm.FindEthTxWithAttempts(etxOtherChain.ID)
+		require.NoError(t, err)
+		require.Len(t, etxOtherChain.EthTxAttempts, 1)
+		attempt = etxOtherChain.EthTxAttempts[0]
+
+		assert.Nil(t, attempt.BroadcastBeforeBlockNum)
+	})
+}
+
+func TestORM_FindEtxAttemptsConfirmedMissingReceipt(t *testing.T) {
```

Comment:
I find it intuitive to have a test per ORM method. This also comes with the benefit of parallelization but may also come with more code.

Reply:
You can also do as such:

```
func TestORM(t *testing.T) {
  t.Run("FindEtxAttemptsConfirmedMissingReceipt", func(t *testing.T){
    ...
  })
  ...
```